src/semidiscretization/semidiscretization.jl

```diff
@@ -155,9 +174,10 @@ end
 Same as [`compute_coefficients`](@ref) but stores the result in `u_ode`.
 """
 function compute_coefficients!(u_ode, func, t, semi::AbstractSemidiscretization)
+    backend = trixi_backend(u_ode)
     u = wrap_array(u_ode, semi)
     # Call `compute_coefficients` defined by the solver
-    compute_coefficients!(u, func, t, mesh_equations_solver_cache(semi)...)
+    compute_coefficients!(backend, u, func, t, mesh_equations_solver_cache(semi)...)
```

Comment:
Should we always pass the `backend` as the first argument, even though it will not be modified?
We should add the convention we choose in the end to https://trixi-framework.github.io/TrixiDocumentation/stable/styleguide/#Conventions